components/equihash/src/tromp.rs

```diff
@@ -30,23 +32,30 @@ extern "C" {
     fn equi_digiteven(eq: *mut CEqui, r: u32, id: u32);
     fn equi_digitK(eq: *mut CEqui, id: u32);
     fn equi_nsols(eq: *const CEqui) -> usize;
-    fn equi_sols(eq: *const CEqui) -> *const *const u32;
+    /// Returns `equi_nsols()` solutions of length `2^K`, in a single memory allocation.
+    fn equi_sols(eq: *const CEqui) -> *const u32;
 }
 
-unsafe fn worker(p: verify::Params, curr_state: &State) -> Vec<Vec<u32>> {
-    // Create solver and initialize it.
-    let eq = equi_new(
-        1,
-        blake2b::blake2b_clone,
-        blake2b::blake2b_free,
-        blake2b::blake2b_update,
-        blake2b::blake2b_finalize,
-    );
```

Comment:
Moving the allocation out of the loop gives me a 5-10% performance boost, but it's optional.